compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1411,6 +1411,13 @@ impl NodeInterner {
     ) -> Result<(), (Span, FileId)> {
         self.trait_implementations.insert(impl_id, trait_impl.clone());
 
+        // Avoid adding error types to impls since they'll conflict with every other type.
+        // We don't need to return an error since we expect an error to already be issued when
+        // the error type is created.
+        if object_type == Type::Error {
+            return Ok(());
+        }
```

Comment:
When a type failed to resolve previously it'd lead to many conflicting trait impl errors on impls which seemingly did not conflict. This removes those errors.